### PR TITLE
Fix [Model endpoints] switching to model/feature set overview is malfunctioning

### PR DESCRIPTION
**Pull request by Taras-Hlukhovetskyi**
https://iguazio.atlassian.net/browse/ML-7582
depends on https://github.com/Taras-Hlukhovetskyi/dashboard-react-controls/pull/10

### CODE REVIEW

File: src/components/ArtifactsActionBar/ArtifactsActionBar.js [hunk 1]
```diff
@@ -35,10 +35,10 @@ import {
   GROUP_BY_NAME,
   GROUP_BY_NONE,
   REQUEST_CANCELED,
-  TAG_FILTER_ALL_ITEMS
+  TAG_FILTER_ALL_ITEMS,
```
Comment (illia-prokopchuk) [inline src/components/ArtifactsActionBar/ArtifactsActionBar.js]: remove comma

File: src/components/FeatureStore/FeatureSets/FeatureSets.js [hunk 2]
```diff
@@ -118,24 +118,28 @@ const FeatureSets = ({
...
+
+          return parsedResult
         }
       })
     },
     [fetchFeatureSets, params.projectName]
   )
 
+  const fetchTags = useCallback(() => {
+    tagAbortControllerRef.current = new AbortController()
```
Comment (illia-prokopchuk) [inline src/components/FeatureStore/FeatureSets/FeatureSets.js:133]: add an empty line

File: src/components/FeatureStore/FeatureVectors/FeatureVectors.js [hunk 3]
```diff
@@ -121,15 +120,26 @@ const FeatureVectors = ({
 
       return fetchFeatureVectors(params.projectName, filters, config).then(result => {
         if (result) {
-          setFeatureVectors(parseFeatureVectors(result))
+          const parsedResult = parseFeatureVectors(result)
```
Comment (illia-prokopchuk) [inline src/components/FeatureStore/FeatureVectors/FeatureVectors.js:123]: add empty line

File: src/components/FeatureStore/FeatureSets/FeatureSets.js [hunk 4]
```diff
@@ -118,24 +118,28 @@ const FeatureSets = ({
...
+    return dispatch(
```
Comment (illia-prokopchuk) [inline src/components/FeatureStore/FeatureSets/FeatureSets.js:135]:
```suggestion
    return dispatch(
      getFilterTagOptions({
        fetchTags: fetchFeatureSetsTags,
        project: params.projectName,
        config: {
          signal: tagAbortControllerRef.current.signal
        }
      }))
```

Comment (illia-prokopchuk) [reply src/components/FeatureStore/FeatureSets/FeatureSets.js:135]: please format

File: src/components/FeatureStore/FeatureVectors/FeatureVectors.js [hunk 5]
```diff
@@ -121,15 +120,26 @@ const FeatureVectors = ({
...
       })
     },
     [fetchFeatureVectors, params.projectName]
   )
 
+  const fetchTags = useCallback(() => {
+    tagAbortControllerRef.current = new AbortController()
+
+    return dispatch(
+      getFilterTagOptions({ fetchTags: fetchFeatureVectorsTags, project: params.projectName, config: {
```
Comment (illia-prokopchuk) [inline src/components/FeatureStore/FeatureVectors/FeatureVectors.js]: please format

File: src/utils/tableRows.util.js [hunk 6]
```diff
@@ -33,3 +35,42 @@ export const isRowExpanded = (parentRef, expandedRowsData, rowItem) => {
     (expandedRowsData && rowItem.data.ui.identifier in expandedRowsData)
   )
 }
+
```
Comment (illia-prokopchuk) [inline src/utils/tableRows.util.js:38]: remove empty line

File: src/utils/tableRows.util.js [hunk 7]
```diff
@@ -33,3 +35,42 @@ export const isRowExpanded = (parentRef, expandedRowsData, rowItem) => {
...
+
+/**
+ * Auto expand table row.
+ * @param {string} name - Row name.
+ * @param {Array} itemsList - List of table items.
+ * @param {string} projectName - Project name.
+ * @param {function} setExpandedRowsData - Function that set expanded date in state.
```
Comment (illia-prokopchuk) [inline src/utils/tableRows.util.js]: date => data

**Review by illia-prokopchuk — CHANGES_REQUESTED** [review]

File: src/utils/tableRows.util.js [hunk 8]
```diff
@@ -33,3 +35,41 @@ export const isRowExpanded = (parentRef, expandedRowsData, rowItem) => {
...
 }
+
+/**
+ * Auto expand table row.
+ * @param {string} name - Row name.
+ * @param {Array} itemsList - List of table items.
+ * @param {string} projectName - Project name.
+ * @param {function} setExpandedRowsData - Function that set expanded data in state.
+ * @param {function} createRowData - Function that create parsed row data.
+ * @param {string} sortBy - Field by witch we will sort items
```
Comment (mavdryk) [inline src/utils/tableRows.util.js]: fix name and description

File: src/utils/tableRows.util.js [hunk 9]
```diff
@@ -33,3 +35,41 @@ export const isRowExpanded = (parentRef, expandedRowsData, rowItem) => {
     (expandedRowsData && rowItem.data.ui.identifier in expandedRowsData)
   )
 }
+
+/**
+ * Auto expand table row.
+ * @param {string} name - Row name.
+ * @param {Array} itemsList - List of table items.
```
Comment (mavdryk) [inline src/utils/tableRows.util.js]: extend Array type

File: src/components/FunctionsPage/Functions.js [hunk 10]
```diff
@@ -503,42 +504,47 @@ const Functions = ({
...
-  useLayoutEffect(() => {
-    if (
-      !functionsAreInitializedRef.current &&
-      (params.funcName || (params.hash && params.hash.includes('@')))
-    ) {
+  const setInitialFilters = useCallback(() => {
+    if (params.funcName || (params.hash && params.hash.includes('@'))) {
       const funcName = params.funcName || params.hash.split('@')[0]
+      const dateFilterValues = getDatePickerFilterValue(datePickerPastOptions, ANY_TIME_DATE_OPTION)
+      const showUntagged = true
```
Comment (mavdryk) [inline src/components/FunctionsPage/Functions.js]: redundant variable

File: src/utils/tableRows.util.js [hunk 11]
```diff
@@ -33,3 +35,41 @@ export const isRowExpanded = (parentRef, expandedRowsData, rowItem) => {
...
+  name,
+  itemsList,
+  projectName,
+  setExpandedRowsData,
+  createRowData,
+  sortListBy
+) => {
+  if (name) {
+    const filteredItems = itemsList.filter(
+      artifact => artifact?.db_key === name || artifact.name === name
```
Comment (mavdryk) [inline src/utils/tableRows.util.js]: this function isn't only for artifacts. Variable name isn't valid

File: src/utils/tableRows.util.js [hunk 12]
```diff
@@ -33,3 +35,41 @@ export const isRowExpanded = (parentRef, expandedRowsData, rowItem) => {
...
+    )
+
+    if (filteredItems.length) {
+      setExpandedRowsData(state => ({
+        ...state,
+        [filteredItems[0]?.ui?.identifier || name]: {
+          content: (sortListBy ? sortListByDate(filteredItems, sortListBy, false) : filteredItems).map(artifact =>
```
Comment (mavdryk) [inline src/utils/tableRows.util.js]: this function isn't only for artifacts. Variable name isn't valid

**Review by mavdryk — CHANGES_REQUESTED** [review]

**Review by mavdryk — APPROVED** [review]

**Pull request closed without merging**